lib/rubocop/magic_comment.rb

```diff
@@ -190,7 +190,7 @@ def frozen_string_literal; end
     class SimpleComment < MagicComment
       # Match `encoding` or `coding`
       def encoding
-        extract(/\#* \b(?:en)?coding: (#{TOKEN})/i)
+        extract(/\A\#.*\b(?:en)?coding: (#{TOKEN})/i)
```

Comment:
I don't think `\A` here is a good idea. If a comment is indented this won't match. That's probably bad style, but I think it will work just fine.

Reply:
I see. I added `\A` just like the pattern match for `frozen_string_literal` magic comment below.

```ruby
def extract_frozen_string_literal
  extract(/\A#\s*frozen[_-]string[_-]literal:\s*(#{TOKEN})\s*\z/i)
end
```

https://github.com/rubocop-hq/rubocop/pull/6117/files#diff-6e19cd40a821113e5df8e1f60f634e7bR206

However, this is not good when indented as pointed out.

For both pattern matching, I'd like to change from `A\#` to `\A\s*\#` to accept indentation.

```diff
% g diff
diff --git a/lib/rubocop/magic_comment.rb b/lib/rubocop/magic_comment.rb
index 60d727c49..8c5a17dcb 100644
--- a/lib/rubocop/magic_comment.rb
+++ b/lib/rubocop/magic_comment.rb
@@ -190,7 +190,7 @@ module RuboCop
     class SimpleComment < MagicComment
       # Match `encoding` or `coding`
       def encoding
-        extract(/\A\#.*\b(?:en)?coding: (#{TOKEN})/i)
+        extract(/\A\s*\#.*\b(?:en)?coding: (#{TOKEN})/i)
       end

       private
@@ -203,7 +203,7 @@ module RuboCop
       # Case-insensitive and dashes/underscores are acceptable.
       # @see https://git.io/vM7Mg
       def extract_frozen_string_literal
-        extract(/\A#\s*frozen[_-]string[_-]literal:\s*(#{TOKEN})\s*\z/i)
+        extract(/\A\s*#\s*frozen[_-]string[_-]literal:\s*(#{TOKEN})\s*\z/i)
       end
     end
   end
```


What do you think about this?

Reply:
Yeah, I think that's the best approach. I'm guessing that some cop about indentation consistency is flagging misindented comments even now, but if not - we should make sure those are flagged by some cop.